File: TP/somadedigitos.c
#include <stdlib.h>
#include <stdio.h>

int somaNumeros(char entrada[]){

    int numero =  atoi(entrada);

    int centena = numero / 100;
    int dezena = (numero % 100) / 10;
    int unidade = numero % 10;

    int soma = centena + dezena + unidade;
    return soma;
}

int main(){
    char entrada[100] ;

    scanf("%s", &entrada);

    while(strcmp(entrada, "FIM") != 0){

        int soma = somaNumeros(entrada);
        printf("%d\n", soma);

        scanf("%s", &entrada);

    }

    return 0;
}

File: TP/SomaDeDigitos.java
import java.util.Scanner;

public class SomaDeDigitos {
    
    public static int soma(int numero){

        int c, d, u;

        c = numero / 100;
        d = (numero % 100)/10;
        u = numero% 10;

        int soma = c + d + u;

        return soma;
    }
    
    public static void main(String[] args) {
        
        Scanner scanner = new Scanner(System.in);
        
        String entrada = scanner.nextLine();

        while (!entrada.equals("FIM")) {
            
            int numero = Integer.parseInt(entrada); 

            int soma = soma(numero);

            System.out.println(soma);

            entrada = scanner.nextLine();

        }
    }
}
